airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -44,9 +38,41 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
                                 final StandardSyncInput syncInput,
                                 final UUID connectionId) {
 
-    StandardSyncOutput syncOutput = replicationActivity.replicate(jobRunConfig, sourceLauncherConfig, destinationLauncherConfig, syncInput);
     final int version = Workflow.getVersion(VERSION_LABEL, Workflow.DEFAULT_VERSION, CURRENT_VERSION);
 
+    final ReplicationActivity replicationActivity;
+    final NormalizationActivity normalizationActivity;
+    final DbtTransformationActivity dbtTransformationActivity;
+    final PersistStateActivity persistActivity;
+
+    /**
+     * The current version calls a new activity to determine which Task Queue to use for other
+     * activities. The previous version doesn't call this new activity, and instead lets each activity
+     * inherit the workflow's Task Queue.
+     */
+    if (version > PREV_VERSION) {
+      final RouteToTaskQueueActivity decideTaskQueueActivity =
+          Workflow.newActivityStub(RouteToTaskQueueActivity.class, ActivityConfiguration.SHORT_ACTIVITY_OPTIONS);
+
+      final String dataPlaneTaskQueue = decideTaskQueueActivity.routeToTaskQueue(connectionId);
```

Comment:
How this part works is not very clear to me, I would somehow expected that it should be part of the existing activities. Might be worth discussing it directly rather than iterating it PR comments.

Reply:
We discussed offline - Jimmy feel free to continue the discussion in the child PR here: https://github.com/airbytehq/airbyte/pull/15905/files